app/ui.js

```diff
@@ -1099,14 +1096,49 @@ var UI;
             // Don't display the connection settings until we're actually disconnected
         },
 
+        reconnect: function() {
+            delete UI.reconnect_callback;
+
+            // if reconnect has been disabled in the meantime, do nothing.
+            if (UI.inhibit_reconnect) {
+                return;
+            }
+
+            UI.connect();
+        },
+
         disconnectFinished: function (rfb, reason) {
+            var reconnect = UI.getSetting('reconnect');
+
+            if ((reconnect === 'true' || reconnect == '1') && !UI.inhibit_reconnect) {
+                document.getElementById("noVNC_transition_text").innerHTML = _("Reconnecting...");
+                document.documentElement.classList.add("noVNC_reconnecting");
+                UI.updateVisualState();
+
+                var delay = parseInt(UI.getSetting('reconnect_delay'));
+                UI.reconnect_callback = setTimeout(UI.reconnect, delay);
+                return;
+            }
+
             if (typeof reason !== 'undefined') {
                 UI.showStatus(reason, 'error');
             }
```

Comment:
We probably still want to show errors. And perhaps not trigger the reconnect in those cases.